src/scripts/main.js

```diff
@@ -1 +1,105 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  let isResolved = false;
+
+  const clickHandler = () => {
+    if (!isResolved) {
+      isResolved = true;
+      clearTimeout(timer);
+      document.removeEventListener('click', clickHandler);
+
+      resolve('First promise was resolved');
+    }
+  };
+
+  document.addEventListener('click', clickHandler);
+
+  const timer = setTimeout(() => {
+    if (!isResolved) {
+      isResolved = true;
+      document.removeEventListener('click', clickHandler);
+
+      reject(new Error('First promise was rejected'));
+    }
+  }, 3000);
+});
+
+firstPromise
+  .then((message) => {
+    const newElement = document.createElement('div');
+
+    newElement.setAttribute('data-qa', 'notification');
+    newElement.classList.add('success');
+    newElement.innerText = message;
+    document.body.appendChild(newElement);
+  })
+  .catch((error) => {
+    const newElement = document.createElement('div');
+
+    newElement.setAttribute('data-qa', 'notification');
+    newElement.classList.add('error');
+    newElement.innerText = error;
+    document.body.appendChild(newElement);
+  });
+
+const secondPromise = new Promise((resolve, reject) => {
+  let isResolved = false;
+
+  const clickHandler = () => {
+    if (!isResolved) {
+      isResolved = true;
+      document.removeEventListener('click', clickHandler);
+
+      resolve('Second promise was resolved');
+    }
+  };
+
+  document.addEventListener('click', clickHandler);
+  document.addEventListener('contextmenu', clickHandler);
```

Comment:
The 'clickHandler' is being added to both 'click' and 'contextmenu' events, but it is only removed from the 'click' event when the promise resolves. You should ensure that the handler is removed from both events to prevent memory leaks.